projects/plugins/jetpack/modules/comments/subscription-modal-on-comment/subscription-modal.js

```diff
@@ -0,0 +1,173 @@
+document.addEventListener( 'DOMContentLoaded', function () {
+	const modal = document.getElementsByClassName( 'jetpack-subscription-modal' )[ 0 ];
+
+	if ( ! modal ) {
+		return;
+	}
+
+	const close = document.getElementsByClassName( 'jetpack-subscription-modal__close' )[ 0 ];
+	const subscribeForms = document.querySelectorAll( '.jetpack-subscription-modal__form' );
+
+	let redirectUrl = '';
+	let subscriptionData = '';
+	let hasLoaded = false;
+
+	function reloadOnCloseSubscriptionModal( customUrl ) {
+		const destinationUrl = customUrl ? new URL( customUrl ) : new URL( redirectUrl );
+
+		// Prevent redirect to external sites.
+		if ( destinationUrl.hostname !== window.location.hostname ) {
+			return;
+		}
+
+		localStorage.setItem( 'jetpack-subscription-modal-goto', destinationUrl.hash );
+
+		// For avoiding Firefox reload, we need to force reload bypassing the cache.
+		window.location.reload( true );
+	}
+
+	function handleSubscriptionModalIframeResult( eventFromIframe ) {
+		if ( eventFromIframe.origin === 'https://subscribe.wordpress.com' && eventFromIframe.data ) {
+			const data = JSON.parse( eventFromIframe.data );
+			const iframeElement = document.querySelector( '.jetpack-subscription-modal__iframe' );
+			if ( data && data.action === 'close' ) {
+				window.removeEventListener( 'message', handleSubscriptionModalIframeResult );
+				iframeElement.src = 'about:blank';
+				reloadOnCloseSubscriptionModal( subscriptionData.url );
+			}
+		}
+	}
+
+	function showSubscriptionIframe( subscriptionData ) {
```

Comment:
I think it's problematic to duplicate these features here, as it becomes even harder to maintain since they're duplicated in many places.

Could we just re-use subscription block's existing functionality here?

Reply:
Most of your suggestions will be fixed by using the subscription block as mentioned here (https://github.com/Automattic/jetpack/pull/34659#discussion_r1442843758f). If you think it's a good idea, I can work on this today.

Reply:
Yeah so as an overall advice I'd look to:

- Consolidate with subscribe block as much as possible
- Consolidate styles/JS with [subscriber modal feature](https://github.com/Automattic/jetpack/tree/83b1208519d32ca61a1f6abad0b7f246a80a928c/projects/plugins/jetpack/modules/subscriptions/subscribe-modal) where you see it makes sense — sometimes it makes sense to copy trivial things, sometimes makes sense to share. It really depends.

All this is fine in follow-ups too, if you'd prefer to get this PR merged already now to get some user-testing/numbers over the weekend. All this is pretty much under-the hood, after all.

Reply:
I just pushed some little fixes and I'm gonna merge it.
I'll create a follow-up to re-use the subscribe block instead of our custom one.

Thanks for all your feedback @simison

Reply:
I asked for a new review since I need it after the last changes 🤦‍♂️